release/models/system/openconfig-system-controlplane.yang

```diff
@@ -199,6 +214,186 @@ module openconfig-system-controlplane {
+                container state {
+                  description
+                    "control-plane protection rsvp counters";
+                  config false;
+                  uses system-controlplane-built-in-limits-traffic-state;
+                }
+              }
+              container l3-routing {
+                description
+                  "Traffic with destination of this device, of all L3 routing
+                  protocols expet bgp and rsvp, which have their own conters
+                  - ISIS, OSPF, LDP, PIM, etc";
+                container state {
+                  description
+                    "control-plane protection L3 routing protocols counters";
+                  config false;
+                  uses system-controlplane-built-in-limits-traffic-state;
+                }
+              }
+              container l3-control {
+                description
+                  "Traffic with destination of this device, of L3 control
+                  protocols: ICMP, VRRP, ARP, IPv6 ND, BFD, MPLS-OAM, IGMP, etc";
+                container state {
+                  description
+                    "control-plane protection L2 control protocols counters";
+                  config false;
+                  uses system-controlplane-built-in-limits-traffic-state;
+                }
+              }
+              container l2-control {
+                description
+                  "Traffic with destination of this device, of L2 control
+                  protocols. E.g. LACP, LLDP, Eth-OAM, BPDU";
+                container state {
+                  description
+                    "control-plane protection L2 control protocols counters";
+                  config false;
+                  uses system-controlplane-built-in-limits-traffic-state;
+                }
+              }
+              container trap-exeption {
+                description
+                  "Traffic with destination other then given device, that are not
+                  forwarded but send to Control Plane for further processing.
+                  Examples are: TTL=1, egress MTU exceeded and DF set, no next-hop
+                  L2 address known - ARP/ND triggering, no match in FIB - ICMP no
+                  route to destination generattion, etc";
+                container state {
+                  description
+                    "control-plane protection trap and exeption packets counters";
+                  config false;
+                  uses system-controlplane-built-in-limits-traffic-state;
+                }
+              }
+              container sample-redirect {
+                description
+                  "Packets with destination other then given device, that are
+                  intercepted to control plane via policy (e.g. P4RT) or samples of
+                  this trafpackets send to control plane for further processing (sFlow,
+                  IPFIX, etc.)";
```

Comment:
This doesn't completely fit the definition of 'built-in' because the policy/config is user provided, but I guess it can still fit here because what the rate limits is (pps,bps) are not (yet) exposed for configuration.

```suggestion
                  "Packets with destination other than the given device which are
                  punted to the control plane via user provided policy (e.g. P4RT) 
                  or configuration such as packet mirroring or packet sampling.  
                  For example, port mirroring, sFlow, IPFIX, etc.";
```